### ISSUE TITLE
skaffold deploy (and even skaffold fix) validates existence of "build->artifacts->context"

### ISSUE THREAD

**Issue by lucasmaj** [GoogleContainerTools/skaffold#4616]
### Expected behavior
Existence of "build->artifacts->context" path has no bearing on "deploy" (nor "fix", nor any other pipeline that does not need it, so "build" only???). When using "deploy" separately from "build", context path (which is used for "build") is not guaranteed  to exist during "deploy". Does "deploy" need context anyway?
### Actual behavior
Error thrown stating that context does not exist. 

> invalid skaffold config: image "some.azurecr.io/myproject" context "./publish/myproject/" does not exist

### Information

- Skaffold version: 1.13.0
- Operating system: Windows and Linux, others unsure
- Contents of skaffold.yaml:

```yaml
apiVersion: skaffold/v1
kind: Config
build:
  artifacts:
  - image: some.azurecr.io/myproject
    context: "./publish/myproject/"
    docker:
      dockerfile: "./path/to/some/Dockerfile"
deploy:
  kubectl:
    manifests:
    - myproject/deployment.yaml
```

### Steps to reproduce the behavior

1. skaffold build (or skaffold fix) above skaffold.yaml
2. To see a working version just comment out "context" or provide existing path.

**Comment by gsquared94** [comment]
Your expected behavior definitely sounds reasonable. I'll discuss in our weekly triage what, if anything, necessitated the existence of a build context always.

**Comment by lucasmaj** [comment]
Also would like to point out that this behavior was introduced in 1.13.0. Our workaround right now is to use v1.12.1.

**Comment by briandealwis** [comment]
This was introduced by #4492.

**Comment by lucasmaj** [comment]
Confirming it is fixed in v1.13.2. Thank you!
